File: Data Structures & Algorithms/contains-duplicate-ii/submission-2.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        seen = set()
        left = right = 0
        while right < len(nums):
            if right - left > k:
                seen.remove(nums[left])
                left += 1
            if nums[right] in seen:
                return True
            seen.add(nums[right])
            right += 1
        return False
        

File: Data Structures & Algorithms/maximum-ascending-subarray-sum/submission-2.cpp
class Solution {
public:
    int maxAscendingSum(vector<int>& nums) {
        int max_sum = 0;
        int curr_sum = 0;
        for (int right = 0; right < nums.size(); right++) {
            if (right == 0) {
                curr_sum += nums[right];
                max_sum = std::max(max_sum, curr_sum);
                continue;
            }
            if (nums[right] > nums[right -1]) {
                curr_sum += nums[right];
                max_sum = std::max(max_sum, curr_sum);
            } else {
                curr_sum = nums[right];
            }
        }
        return max_sum; 
    }
};